src/Umbraco.Core/Services/ContentEditingService.cs

```diff
@@ -240,13 +244,71 @@ public async Task<Attempt<ContentUpdateResult, ContentEditingOperationStatus>> U
     }
 
     public async Task<Attempt<IContent?, ContentEditingOperationStatus>> MoveToRecycleBinAsync(Guid key, Guid userKey)
-        => await HandleMoveToRecycleBinAsync(key, userKey);
+    {
```

Comment:
Unsure if this is the right solution, the methods from the base class aren't abstract, so we could still override them theoretically.

This is basically just copied entirely from the base class with only this snippet addet in there: 

```
        if (_contentSettings.DisableUnpublishWhenReferenced && _relationService.IsRelated(content.Id))
        {
            return Attempt.FailWithStatus<IContent?, ContentEditingOperationStatus>(ContentEditingOperationStatus.CannotMoveToRecycleBinWhenReferenced, content);
        }
```